nusacrowd/nusa_datasets/cod/cod.py

```diff
@@ -0,0 +1,142 @@
+English Schema-Guided Dialogue (SGD; Shah et al., 2018; Rastogi et al., 2020) dataset is automatically sampled and mapped into outlines. The outlines are then paraphrased and adapted to the local target domain by human subjects.
+"""
+
+_HOMEPAGE = "https://github.com/cambridgeltl/COD"
+
+_LICENSE = "Unknown"
+
+_URLS = {
+    _DATASETNAME: {"validation": "https://raw.githubusercontent.com/cambridgeltl/COD/main/id_dev.json", "test": "https://raw.githubusercontent.com/cambridgeltl/COD/main/id_test.json"},
+}
+
+_SUPPORTED_TASKS = [Tasks.DIALOGUE_SYSTEM]
+
+_SOURCE_VERSION = "1.0.0"
+
+_NUSANTARA_VERSION = "1.0.0"
+
+
+class NewDataset(datasets.GeneratorBasedBuilder):
+    """Cross-lingual Outline-based Dialogue (COD) is a dataset comprises manually generated, localised, and cross-lingually aligned Task-Oriented-Dialogue (TOD) data which served as the source of dialogue prompts."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    NUSANTARA_VERSION = datasets.Version(_NUSANTARA_VERSION)
+
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="cod_source",
+            version=SOURCE_VERSION,
+            description="Cross-lingual Outline-based Dialogue (COD) source schema",
+            schema="source",
+            subset_id="cod",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "cod_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "index": datasets.Value("string"),
+                    "dialogue_id": datasets.Value("string"),
+                    "services": [datasets.Value("string")],
+                    "turns": [
+                        {
+                            "speaker": datasets.Value("string"),
+                            "utterance": datasets.Value("string"),
+                            "frames": [
+                                {
+                                    "actions": [{"act": datasets.Value("string"), "slot": datasets.Value("string"), "values": [datasets.Value("string")]}],
+                                    "service": datasets.Value("string"),
+                                    "slots": [
+                                        {
+                                            "exclusive_end": datasets.Value("int32"),
+                                            "slot": datasets.Value("string"),
+                                            "start": datasets.Value("int32"),
+                                        }
+                                    ],
+                                    "state": {"active_intent": datasets.Value("string"), "requested_slots": [datasets.Value("string")], "slot_values": [{"slot": datasets.Value("string"), "values": [datasets.Value("string")]}]},
+                                }
+                            ],
+                        }
+                    ],
+                }
+            )
```

Comment:
I follow the original JSON schema. The only differences are:
- Adding `index` param
- Reformat turns.frames.state.slot_values from map[slot]values to list of {slot, values}